lib/SILGen/SILGenConstructor.cpp

```diff
@@ -914,6 +914,10 @@ static Type getInitializationTypeInContext(
     if (auto originalProperty = singleVar->getOriginalWrappedProperty()) {
       if (originalProperty->isPropertyWrapperInitializedWithInitialValue())
         interfaceType = originalProperty->getValueInterfaceType();
+        if (originalProperty->
+            isInnermostPropertyWrapperInitUsesEscapingAutoClosure()) {
+          interfaceType = FunctionType::get({}, interfaceType);
```

Comment:
This check for `isInnermostPropertyWrapperInitUsesEscapingAutoClosure` followed by `FunctionType::get()` occurs in a few places, and it would be really easy to forget to do this adjustment anywhere else in the code. Can you add a new API on `VarDecl` that gets the interface type for wrapped value initialization, which does the `getValueInterfaceType()` and (now) the `FunctionType::get()`?